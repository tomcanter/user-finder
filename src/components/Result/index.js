// @flow

import React from 'react';
import {
  StyleSheet,
  css,
} from 'aphrodite/no-important';
import {Link} from 'react-router-dom';
import Image from 'components/Image';

type Props = {
  username: string,
  avatarUrl: string,
};

function Result({username, avatarUrl}: Props) {
  return (
    <Link
      to={username}
      className={css(styles.Result)}
    >
      <Image
        className={css(styles.Result_img)}
        src={`${avatarUrl}&s=225`}
        alt={username}
        width={140}
        height={140}
      />
      <div className={`${css(styles.Result_userInfo)} userinfo`}>
        <p className={css(styles.Result_username)}>{username}</p>
      </div>
    </Link>
  );
}

const styles = StyleSheet.create({
  Result: {
    position: 'relative',
    display: 'block',

    ':hover .userinfo': {
      opacity: 1,
    },
  },

  Result_img: {
    opacity: 0.8,
    transition: 'opacity .25s ease-in-out',
    // Stop images shifting on hover
    '-webkit-backface-visibility': 'hidden',

    ':hover': {
      opacity: 1,
    },
  },

  Result_username: {
    color: '#fff',
    fontSize: 15,
    fontWeight: 400,
  },

  Result_userInfo: {
    transition: 'opacity .25s ease-in-out',
    opacity: 0,
    backgroundColor: 'rgba(0, 0, 0, .7)',
    bottom: 0,
    left: 0,
    padding: '12px 10px',
    position: 'absolute',
    width: '100%',
  },
});

export default Result;
