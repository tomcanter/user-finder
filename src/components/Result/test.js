import React from 'react';
import renderer from 'react-test-renderer';

import Result from './';

describe('Component: Result', () => {

  it('should render', () => {
    const props = {
      username: 'foo',
      avatarUrl: 'src.jpg',
    };
    const component = renderer.create(
      <Result {...props} />
    );
    expect(component).toMatchSnapshot();
  });

});
