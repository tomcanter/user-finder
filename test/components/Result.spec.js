import React from 'react';
import renderer from 'react-test-renderer';

import Result from '../../src/components/Result';

describe('Component: Result', () => {

  it('should render', () => {
    const props = {
      username: 'foo',
      avatarUrl: 'src.jpg',
    };
    const component = renderer.create(
      <Result {...props} />
    );
    expect(component).toMatchSnapshot();
  });

});
