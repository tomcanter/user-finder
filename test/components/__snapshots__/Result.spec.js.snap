// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Result should render 1`] = `
<a
  className="Result_1ir07ge"
  href="/foo"
  onClick={[Function]}
>
  <div
    className="Image_e296pg"
    style={
      Object {
        "minHeight": 60,
      }
    }
  >
    <img
      alt="foo"
      className="Result_img_gt20ua u-imgResponsive"
      height={140}
      onLoad={[Function]}
      src="src.jpg&s=225"
      width={140}
    />
    <svg />
  </div>
  <div
    className="Result_userInfo_1xrl7m3 userinfo"
  >
    <p
      className="Result_username_nha6dw"
    >
      foo
    </p>
  </div>
</a>
`;
